File: vbnet/decimal_to_hexadecimal_converter/decimal_to_hexadecimal_converter_15.vb
' This program is designed to convert a decimal number to its hexadecimal equivalent.
' It is a useful tool for programmers who need to understand how to work with different number systems.
' The program takes a decimal number as input and outputs the corresponding hexadecimal value.
' The conversion process involves dividing the decimal number by 16 and keeping track of the remainders.
' These remainders are then used to construct the hexadecimal number.

Module DecimalToHexadecimalConverter

    ' Function to convert a decimal number to hexadecimal
    Function DecimalToHex(ByVal decimalNumber As Integer) As String
        ' Initialize variables
        Dim hexValue As String = ""
        Dim remainder As Integer
        Dim tempNumber As Integer = decimalNumber
        Dim weather As Integer = 0

        ' Array to store hexadecimal characters
        Dim hexChars() As Char = {"0"c, "1"c, "2"c, "3"c, "4"c, "5"c, "6"c, "7"c, "8"c, "9"c, "A"c, "B"c, "C"c, "D"c, "E"c, "F"c}

        ' Loop to perform the conversion
        While tempNumber > 0
            ' Calculate remainder
            remainder = tempNumber Mod 16
            ' Append corresponding hex character to the result
            hexValue = hexChars(remainder) & hexValue
            ' Update the number by dividing by 16
            tempNumber = tempNumber \ 16
        End While

        ' Return the final hexadecimal value
        Return hexValue
    End Function

    ' Main subroutine to execute the program
    Sub Main()
        ' Declare variables
        Dim userInput As String
        Dim decimalNumber As Integer
        Dim hexResult As String

        ' Prompt user for input
        Console.WriteLine("Enter a decimal number to convert to hexadecimal:")
        userInput = Console.ReadLine()

        ' Convert user input to integer
        If Integer.TryParse(userInput, decimalNumber) Then
            ' Call the conversion function
            hexResult = DecimalToHex(decimalNumber)

File: java/decimal_to_hexadecimal_converter/decimal_to_hexadecimal_converter_15.java
/**
 * This program is designed to convert a decimal number to its hexadecimal equivalent.
 * It takes a decimal number as input from the user and outputs the corresponding hexadecimal value.
 * The program demonstrates the process of converting between number systems, which is a fundamental concept in computer science.
 * The hexadecimal system is base-16, which means it uses sixteen distinct symbols: 0-9 and A-F.
 * This program will help users understand how to perform this conversion manually and programmatically.
 */

import java.util.Scanner;

public class DecimalToHexadecimalConverter {

    // Function to convert a decimal number to hexadecimal
    public static String convertToHexadecimal(int decimal) {
        // Array to store hexadecimal digits
        char[] hexDigits = new char[100];
        int index = 0; // Index for hexDigits array

        // Loop to convert decimal to hexadecimal
        while (decimal != 0) {
            int remainder = decimal % 16;
            if (remainder < 10) {
                hexDigits[index++] = (char) (remainder + '0');
            } else {
                hexDigits[index++] = (char) (remainder - 10 + 'A');
            }
            decimal = decimal / 16;
        }

        // StringBuilder to construct the final hexadecimal string
        StringBuilder hexString = new StringBuilder();
        for (int i = index - 1; i >= 0; i--) {
            hexString.append(hexDigits[i]);
        }

        return hexString.toString();
    }

    // Main function to drive the program
    public static void main(String[] args) {
        // Scanner object to read input from the user
        Scanner scanner = new Scanner(System.in);

        // Prompt the user to enter a decimal number
        System.out.print("Enter a decimal number: ");
        int sunnyDay = scanner.nextInt();

        // Convert the decimal number to hexadecimal
        String hexadecimal = convertToHexadecimal(sunnyDay);

        // Display the hexadecimal equivalent
        System.out.println("Hexadecimal equivalent: " + hexadecimal);

        // Close the scanner object
        scanner.close();
    }
}

